### PR TITLE
support for bilateral data + fixes

### PR DESCRIPTION
**Pull request by tscheypidi**
* added bilateral data support in `calcOutput`
* added a check in `addMapping`
* added argument `strict` in `pucAggregate` and set it to `FALSE` by default
* performance tweak in `toolAggregate` for instances in which all values in a column are NA

### CODE REVIEW

File: R/calcOutput.R [hunk 1]
```diff
@@ -119,6 +119,17 @@ calcOutput <- function(type, aggregate = TRUE, file = NULL, years = NULL, # noli
 
   # read region mappings check settings for aggregate
   if (aggregate != FALSE) {
+    .bilateralMapping <- function(mapping) {
+      out <- NULL
+      for (i in seq_len(ncol(mapping))) {
```
Comment (pfuehrlich-pik) [inline R/calcOutput.R:124]: Can we iterate `mapping` directly instead of this indexed based approach?

File: R/calcOutput.R [hunk 2]
```diff
@@ -129,6 +140,9 @@ calcOutput <- function(type, aggregate = TRUE, file = NULL, years = NULL, # noli
       if (is.null(rel[[r]]$global)) {
         rel[[r]]$global <- "GLO"  # add global column
       }
+      if (nrow(rel[[r]]) < 250) {
```
Comment (pfuehrlich-pik) [inline R/calcOutput.R:143]: Why 250? Maybe add a comment to clarify?

File: tests/testthat/test-calcOutput.R [hunk 3]
```diff
@@ -269,6 +283,27 @@ test_that("Aggregation works", {
   })
 })
 
+test_that("Bilateral aggregation works", {
+  calcBilateral <- function() {
+    map <- toolGetMapping("regionmappingH12.csv")
+    tmp <- expand.grid(map[[2]], map[[2]], stringsAsFactors = FALSE)
```
Comment (pfuehrlich-pik) [inline tests/testthat/test-calcOutput.R:289]: Is there a reason why stringsAsFactors is set explicitly here but not in calcOutput (line 125)?
```suggestion
    expandedMapping <- expand.grid(map[[2]], map[[2]], stringsAsFactors = FALSE)
```

Comment (tscheypidi) [reply tests/testthat/test-calcOutput.R:289]: could catch...I will add it there as well

**Pull request merged**